mpitests/test_apply.py

```diff
@@ -0,0 +1,480 @@
+        def circuit_mpi():
+            qml.QubitStateVector(state_vector, wires=range(num_wires))
+            operation(wires=Wires)
+            return qml.state()
+
+        local_state_vector = circuit_mpi()
+
+        assert np.allclose(local_state_vector, local_expected_output_cpu, atol=tol, rtol=0)
+
+    test_gates_qnode_param = [
+        (qml.PhaseShift, [0.1], [0]),
+        (qml.PhaseShift, [0.1], [numQubits - 1]),
+        (qml.RX, [0.2], [0]),
+        (qml.RX, [0.2], [numQubits - 1]),
+        (qml.RY, [0.3], [0]),
+        (qml.RY, [0.3], [numQubits - 1]),
+        (qml.RZ, [0.4], [0]),
+        (qml.RZ, [0.4], [numQubits - 1]),
+        (qml.Rot, [0.1, 0.2, 0.3], [0]),
+        (qml.Rot, [0.1, 0.2, 0.3], [numQubits - 1]),
+        (qml.CRX, [0.1], [0, 1]),
+        (qml.CRX, [0.1], [0, numQubits - 1]),
+        (qml.CRX, [0.1], [numQubits - 2, numQubits - 1]),
+        (qml.CRY, [0.2], [0, 1]),
+        (qml.CRY, [0.2], [0, numQubits - 1]),
+        (qml.CRY, [0.2], [numQubits - 2, numQubits - 1]),
+        (qml.CRZ, [0.3], [0, 1]),
+        (qml.CRZ, [0.3], [0, numQubits - 1]),
+        (qml.CRZ, [0.3], [numQubits - 2, numQubits - 1]),
+        (qml.ControlledPhaseShift, [0.4], [0, 1]),
+        (qml.ControlledPhaseShift, [0.4], [0, numQubits - 1]),
+        (qml.ControlledPhaseShift, [0.4], [numQubits - 2, numQubits - 1]),
+        (qml.SingleExcitation, [0.5], [0, 1]),
+        (qml.SingleExcitation, [0.5], [0, numQubits - 1]),
+        (qml.SingleExcitation, [0.5], [numQubits - 2, numQubits - 1]),
+        (qml.SingleExcitationMinus, [0.6], [0, 1]),
+        (qml.SingleExcitationMinus, [0.6], [0, numQubits - 1]),
+        (qml.SingleExcitationMinus, [0.6], [numQubits - 2, numQubits - 1]),
+        (qml.SingleExcitationPlus, [0.7], [0, 1]),
+        (qml.SingleExcitationPlus, [0.7], [0, numQubits - 1]),
+        (qml.SingleExcitationPlus, [0.7], [numQubits - 2, numQubits - 1]),
+        (qml.IsingXX, [0.8], [0, 1]),
+        (qml.IsingXX, [0.8], [0, numQubits - 1]),
+        (qml.IsingXX, [0.8], [numQubits - 2, numQubits - 1]),
+        (qml.IsingYY, [0.9], [0, 1]),
+        (qml.IsingYY, [0.9], [0, numQubits - 1]),
+        (qml.IsingYY, [0.9], [numQubits - 2, numQubits - 1]),
+        (qml.IsingZZ, [0.1], [0, 1]),
+        (qml.IsingZZ, [0.1], [0, numQubits - 1]),
+        (qml.IsingZZ, [0.1], [numQubits - 2, numQubits - 1]),
+        (qml.CRot, [0.1, 0.2, 0.3], [0, 1]),
+        (qml.CRot, [0.1, 0.2, 0.3], [0, numQubits - 1]),
+        (qml.CRot, [0.1, 0.2, 0.3], [numQubits - 2, numQubits - 1]),
+        (qml.DoubleExcitation, [0.1], [0, 1, 2, 3]),
+        (qml.DoubleExcitation, [0.1], [0, 1, numQubits - 2, numQubits - 1]),
+        (qml.DoubleExcitationPlus, [0.2], [0, 1, 2, 3]),
+        (qml.DoubleExcitationPlus, [0.2], [0, 1, numQubits - 2, numQubits - 1]),
+        (qml.DoubleExcitationMinus, [0.3], [0, 1, 2, 3]),
+        (qml.DoubleExcitationMinus, [0.3], [0, 1, numQubits - 2, numQubits - 1]),
+    ]
```

Comment:
You can probably, with a few changes, update this test to be double-parametrized.